### ISSUE TITLE
How to keep deleted posts

### ISSUE THREAD

**Issue by woodenphone** [friendlyanon/4chan-thread-archiver#2]
You need to parse the different versions of the page to find each post, and rebuild the page with all the posts included.
You could either:
save each version of the page you retrieve, then combine them into a single page using the HTML from any of the remote version to fill in the HTML before and after the section with the posts when the page 404s.
 save the most recent HTML and all the extracted posts to disk and recombine on 404, 
Parse the remote HTML and the local HTML if it exists, add any missing posts to the HTML and save the new HTML with posts from both versions to disk, repeating this each time you download the html for a page.

Basically:
Load page from remote server
Check for 404
Parse for images and posts (Regex, DOM, etc)
Save any new images
Load local HTML if it exists and then parse that
Join the extracted posts into one html page, removing duplicated posts
Save the new html with all posts to disk
repeat as needed

**Comment by friendlyanon** [comment]
Fixing this issue would require the "core" of the script (them giant regexes in the middle) to be reworked since the only solution I can think of is creating some sort of cache of all the posts in a thread and then parse the HTML from there. My only thoughts on this that it can be done with some effort, but I'd have to think of a format that I can use to convert both online and offline threads into and parse the offline HTML from there.

Now here comes my laziness... so yeah
